File: js/src/maplestory/api/dto/character/characterHexaMatrixStatDto.ts
import { CharacterHexaMatrixStatCoreDto } from './characterHexaMatrixStatCoreDto';
import { CharacterHexaMatrixStatBody } from '../../response/character/characterHexaMatrixStatBody';

/**
 * 캐릭터 HEXA 스탯 코어 정보
 */
class CharacterHexaMatrixStatDto {
  /**
   * 조회 기준일
   */
  date: Date | null;

  /**
   * 캐릭터 직업
   */
  characterClass: string | null;

  /**
   * HEXA 스탯 I 코어 정보
   */
  characterHexaStatCore: CharacterHexaMatrixStatCoreDto[];

  /**
   * HEXA 스탯 II 코어 정보
   */
  characterHexaStatCore2: CharacterHexaMatrixStatCoreDto[];

  /**
   * HEXA 스탯 III 코어 정보
   */
  characterHexaStatCore3: CharacterHexaMatrixStatCoreDto[];

  /**
   * 프리셋 HEXA 스탯 I 코어 정보
   */
  presetHexaStatCore: CharacterHexaMatrixStatCoreDto[];

  /**
   * 프리셋 HEXA 스탯 II 코어 정보
   */
  presetHexaStatCore2: CharacterHexaMatrixStatCoreDto[];

  /**
   * 프리셋 HEXA 스탯 III 코어 정보
   */
  presetHexaStatCore3: CharacterHexaMatrixStatCoreDto[];

  constructor(obj: CharacterHexaMatrixStatBody) {
    const {
      date,
      character_class,
      character_hexa_stat_core,
      character_hexa_stat_core_2,
      character_hexa_stat_core_3,
      preset_hexa_stat_core,
      preset_hexa_stat_core_2,
      preset_hexa_stat_core_3,
    } = obj;

    this.date = date ? new Date(date) : null;
    this.characterClass = character_class;
    this.characterHexaStatCore = character_hexa_stat_core?.map(
      (core) => new CharacterHexaMatrixStatCoreDto(core),
    ) ?? [];
    this.characterHexaStatCore2 = character_hexa_stat_core_2?.map(
      (core) => new CharacterHexaMatrixStatCoreDto(core),
    ) ?? [];
    this.characterHexaStatCore3 = character_hexa_stat_core_3?.map(
      (core) => new CharacterHexaMatrixStatCoreDto(core),
    ) ?? [];
    this.presetHexaStatCore = preset_hexa_stat_core?.map(
      (core) => new CharacterHexaMatrixStatCoreDto(core),
    ) ?? [];
    this.presetHexaStatCore2 = preset_hexa_stat_core_2?.map(
      (core) => new CharacterHexaMatrixStatCoreDto(core),
    ) ?? [];
    this.presetHexaStatCore3 = preset_hexa_stat_core_3?.map(
      (core) => new CharacterHexaMatrixStatCoreDto(core),
    ) ?? [];
  }
}

export { CharacterHexaMatrixStatDto };

File: java/src/main/java/dev/spiralmoon/maplestory/api/msea/dto/character/CharacterHexaMatrixStatDTO.java
package dev.spiralmoon.maplestory.api.msea.dto.character;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonSetter;
import com.fasterxml.jackson.annotation.Nulls;
import dev.spiralmoon.maplestory.api.Utils;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import java.time.ZonedDateTime;
import java.util.List;

/**
 * Character HEXA Matrix configured HEXA stats information
 */
@AllArgsConstructor
@NoArgsConstructor
@Data
@ToString
public class CharacterHexaMatrixStatDTO implements dev.spiralmoon.maplestory.api.common.dto.character.CharacterHexaMatrixStatDTO<
        CharacterHexaMatrixStatCoreDTO> {

    /**
     * Reference date for query (SGT, daily data with hours and minutes set to 0)
     */
    @JsonProperty("date")
    private String date;

    /**
     * Character job
     */
    @JsonProperty("character_class")
    private String characterClass;

    /**
     * Information for HEXA Stat Node I
     */
    @JsonProperty("character_hexa_stat_core")
    @JsonSetter(nulls = Nulls.AS_EMPTY)
    private List<CharacterHexaMatrixStatCoreDTO> characterHexaStatCore;

    /**
     * Information for HEXA Stat Node II
     */
    @JsonProperty("character_hexa_stat_core_2")
    @JsonSetter(nulls = Nulls.AS_EMPTY)
    private List<CharacterHexaMatrixStatCoreDTO> characterHexaStatCore2;

    /**
     * Information for HEXA Stat Node III
     */
    @JsonProperty("character_hexa_stat_core_3")
    @JsonSetter(nulls = Nulls.AS_EMPTY)
    private List<CharacterHexaMatrixStatCoreDTO> characterHexaStatCore3;

    /**
     * Preset HEXA Stat Node I information
     */
    @JsonProperty("preset_hexa_stat_core")
    @JsonSetter(nulls = Nulls.AS_EMPTY)
    private List<CharacterHexaMatrixStatCoreDTO> presetHexaStatCore;

    /**
     * Preset HEXA Stat Node II information
     */
    @JsonProperty("preset_hexa_stat_core_2")
    @JsonSetter(nulls = Nulls.AS_EMPTY)
    private List<CharacterHexaMatrixStatCoreDTO> presetHexaStatCore2;

    /**
     * Preset HEXA Stat Node III information
     */
    @JsonProperty("preset_hexa_stat_core_3")
    @JsonSetter(nulls = Nulls.AS_EMPTY)
    private List<CharacterHexaMatrixStatCoreDTO> presetHexaStatCore3;

    /**
     * Reference date for query (SGT, daily data with hours and minutes set to 0)
     */
    public ZonedDateTime getDate() {
        return date != null
                ? Utils.toZonedDateTime(this.date)
                : null;
    }
}
